include/xgboost/gbm.h

```diff
@@ -154,6 +154,12 @@ class GradientBooster {
       GenericParameter const* gparam,
       const std::vector<std::shared_ptr<DMatrix> >& cache_mats,
       bst_float base_margin);
+
+  static void AssertGPUSupport() {
```

Comment:
Again, we should move to a helper function and do checks and validation against permutation of configurations